uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/LayoutRESTController.java

```diff
@@ -183,4 +148,47 @@ public ModelAndView getRESTController(
 
         return null;
     }
+
+    private void setAltMaxURL(
+            HttpServletRequest request,
+            NamedNodeMap attributes,
+            IPortletDefinition def,
+            LayoutPortlet portlet) {
+        final String alternativeMaximizedLink = def.getAlternativeMaximizedLink();
+        if (alternativeMaximizedLink != null) {
+            portlet.setUrl(alternativeMaximizedLink);
+            portlet.setAltMaxUrl(true);
+        } else {
+            // get the maximized URL for this portlet
+            final IPortalUrlBuilder portalUrlBuilder =
+                    urlProvider.getPortalUrlBuilderByLayoutNode(
+                            request, attributes.getNamedItem("ID").getNodeValue(), UrlType.RENDER);
+            final IPortletWindowId targetPortletWindowId =
+                    portalUrlBuilder.getTargetPortletWindowId();
+            if (targetPortletWindowId != null) {
+                final IPortletUrlBuilder portletUrlBuilder =
+                        portalUrlBuilder.getPortletUrlBuilder(targetPortletWindowId);
+                portletUrlBuilder.setWindowState(WindowState.MAXIMIZED);
+            }
+            portlet.setUrl(portalUrlBuilder.getUrlString());
+            portlet.setAltMaxUrl(false);
+        }
+    }
+
+    private NodeList getNodeList(
+            @RequestParam(value = "tab", required = false) String tab, Document document) {
```

Comment:
@bjagg you should review this part, @RequestParam on private method... an abusive copy/paste

Reply:
fixed!